Source/MockCapture.cs

```diff
@@ -0,0 +1,131 @@
+//    and/or other materials provided with the distribution.
+
+//    * Neither the name of Clarius Consulting, Manas Technology Solutions or InSTEDD nor the 
+//    names of its contributors may be used to endorse 
+//    or promote products derived from this software 
+//    without specific prior written permission.
+
+//THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+//CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+//INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+//MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+//DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+//CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+//SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+//BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+//SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+//INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+//WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+//NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+//OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+//SUCH DAMAGE.
+
+//[This is the BSD license, see
+// http://www.opensource.org/licenses/bsd-license.php]
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Moq
+{
+    /// <summary>
+    /// Captures a mock parameter.
+    /// </summary>
+    /// <example>
+    /// Arrange code:
+    /// <code>
+    /// var capture = new MockCapture{string}();
+    /// mock.Setup(x => x.DoSomething(capture.AsParameter()));
+    /// </code>
+    /// Assert code:
+    /// <code>
+    /// Assert.Equal("Hello!", capture.Last);
+    /// </code>
+    /// </example>
+    public class MockCapture<T>
+    {
+        private readonly List<T> _capturedValues = new List<T>();
+        private Action<T> _callback;
+        private Func<T, T> _copier;
+
+        /// <summary>
+        /// Gets all the captured parameter values.
+        /// </summary>
+        public IEnumerable<T> All { get { return _capturedValues; } }
+
+        /// <summary>
+        /// Gets the last captured parameter value.
+        /// </summary>
+        public T Last { get { return _capturedValues.Last(); } }
+        
+        /// <summary>
+        /// Gets the single captured parameter value.
+        /// </summary>
+        public T Single { get { return _capturedValues.Single(); } }
+
+        /// <summary>
+        /// Gets the parameter to use in the setup expression.
+        /// </summary>
+        /// <returns>The setup expression parameter</returns>
+        public T CaptureAny()
+        {
+            return Match.Create(value => CaptureParameter(value), () => It.IsAny<T>());
+        }
+
+        /// <summary>
+        /// Gets the parameter to use in the setup expression.
+        /// </summary>
+        /// <returns>The setup expression parameter</returns>
+        public T Capture(Expression<Func<T, bool>> match)
```

Comment:
Ok, no problem. This is the name I wanted to use in the first place but I changed it to `match` in order to use the name of the `It.Is` method parameter.